common/utilities/string/trim-newlines.h

```diff
@@ -0,0 +1,60 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+
+namespace utilities {
+namespace string {
+
+
+// Prepare a block of text for word-wrap by removing newlines:
+//     - Remove trailing spaces from lines
+//     - Multiple newlines stay as-is, so still separate paragraphs
+//     - Newlines preceded by colons stay
+//     - Newlines not preceded or followed by anything (beginning/end) stay
+//     - All other newlines replaced by spaces
+// Example: (\n added just for verbosity)
+//     First line \n       // extra space removed
+//     second line\n       // joined into first line
+//     \n                  // stays
+//     third line: \n      // trimmed; stays
+//         fourth\n        // trailing \n stays
+// Becomes:
+//     First line second line\n
+//     \n
+//     third line:\n
+//         fourth\n
+inline std::string trim_newlines( std::string s )
+{
+    char const * const base = s.c_str();
```

Comment:
Can you add some comments to the code? What each section does? 
maybe about the variables? you got s, base, src and dest , hard to understand what are their roles.